src/cmd/services/m3coordinator/downsample/downsampler.go

```diff
@@ -58,7 +58,7 @@ type SamplesAppenderResult struct {
 type SampleAppenderOptions struct {
 	Override      bool
 	OverrideRules SamplesAppenderOverrideRules
-	MetricType    ts.MetricType
+	MetricType    ts.M3MetricType
```

Comment:
Well, we can. But I renamed it in order to match `PromMetricType` in https://github.com/m3db/m3/pull/2628/files/aeb8036681303b0de6312eb47006d49f59a4a631#diff-0535bd9002d5b34759706aebeec1ee46
I believe having `PromMetricType` and just `MetricType` at the same scope/level would be confusing.

Reply:
My thinking was that `MetricType` is implicitly scoped to `M3MetricType` just because we're in the M3 project, so all types can be considered `M3Type` by default

Reply:
I wish it were so simple :) Unfortunately we bring in a protobuf with a different `MetricType` from Prometheus project which makes this whole point moot.